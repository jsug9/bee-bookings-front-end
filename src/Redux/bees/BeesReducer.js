/* eslint-disable no-param-reassign */
import axios from 'axios';
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const GET_BEES = 'Bees/GET_BEES';
const GET_BEE_DETAILS = 'Bees/GET_BEE_DETAILS';
const ADD_BEE = 'Bees/ADD_BEE';
const DELETE_BEE = 'Bees/DELETE_BEE';

const getBeesEndpoint = 'https://bee-store.herokuapp.com/api/v1/items';

const initialState = {
  allBees: [],
  beeDetails: {},
  isLoading: false,
};

export const getAllBees = createAsyncThunk(GET_BEES, async () => {
  const { data } = await axios.get(getBeesEndpoint);
  return data;
});

export const getBeeDetails = createAsyncThunk(GET_BEE_DETAILS, async (id) => {
  const { data } = await axios.get(`${getBeesEndpoint}/${id}`);
  return data;
});

export const addBee = createAsyncThunk(ADD_BEE, async (bee) => {
  axios.post(getBeesEndpoint, bee);
  const { data } = await axios.get(getBeesEndpoint);
  return data;
});

export const deleteBee = createAsyncThunk(DELETE_BEE, async (beeId) => {
  axios.delete(`${getBeesEndpoint}/${beeId}`);
  const { data } = await axios.get(getBeesEndpoint);
  return data;
});

const beesSlice = createSlice({
  name: 'Bees',
  initialState,
  reducers: {},
  extraReducers: {
    [getAllBees.fulfilled]: (state, action) => {
      state.allBees = action.payload.sort((a, b) => a.id - b.id);
      state.isLoading = false;
    },
    [getAllBees.rejected]: (state) => {
      state.allBees = [];
      state.isLoading = false;
    },
    [getAllBees.pending]: (state) => {
      state.allBees = [];
      state.isLoading = true;
    },
    [getBeeDetails.fulfilled]: (state, action) => {
      state.beeDetails = action.payload;
      state.isLoading = false;
    },
    [getBeeDetails.rejected]: (state) => {
      state.beeDetails = {};
      state.isLoading = false;
    },
    [getBeeDetails.pending]: (state) => {
      state.beeDetails = {};
      state.isLoading = true;
    },
    [addBee.fulfilled]: (state, action) => {
      state.allBees = action.payload;
      state.isLoading = false;
    },
    [addBee.rejected]: (state) => {
      state.allBees = [];
      state.isLoading = false;
    },
    [addBee.pending]: (state) => {
      state.allBees = [];
      state.isLoading = true;
    },
    [deleteBee.fulfilled]: (state, action) => {
      state.allBees = action.payload;
      state.isLoading = false;
    },
    [deleteBee.rejected]: (state) => {
      state.allBees = [];
      state.isLoading = false;
    },
    [deleteBee.pending]: (state) => {
      state.allBees = [];
      state.isLoading = true;
    },
  },
});

export default beesSlice.reducer;
