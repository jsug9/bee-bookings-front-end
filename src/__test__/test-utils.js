/* eslint-disable react/prop-types */
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import setupStore from '../Redux/testStore';

function renderWithProviders(
  ui,
  {
    preloadedState,
    store = setupStore(preloadedState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Router><Provider store={store}>{children}</Provider></Router>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export default renderWithProviders;
