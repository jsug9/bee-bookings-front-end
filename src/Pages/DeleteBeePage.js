import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  Select, MenuItem, InputLabel, FormControl, Button,
} from '@mui/material';
import RemoveCircleIcon from '@mui/icons-material/RemoveCircle';
import { useNavigate } from 'react-router-dom';
import { getAllBees, deleteBee } from '../Redux/bees/BeesReducer';
import '../Styles/BeeForm.scss';
import '../Styles/ReservationForm.scss';

const DeleteBeePage = () => {
  const navigate = useNavigate();
  const [beeId, setBeeId] = useState('');
  const dispatch = useDispatch();
  const bees = useSelector((state) => state.bees.allBees);
  const [empty, setEmpty] = useState(true);

  useEffect(() => {
    dispatch(getAllBees());
  }, []);

  useEffect(() => {
    bees.forEach((bee) => {
      if (bee.id > 4) {
        setEmpty(false);
      }
    });
  }, [bees]);

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(deleteBee(beeId));
    navigate('/');
  };

  return (
    <div className="reservation-form-container">
      <form className="bee-form reservation-form" method="post" onSubmit={handleSubmit}>
        <h1 style={{ 'margin-bottom': '30px' }}>Delete a Bee</h1>
        <FormControl>
          <InputLabel id="bee-label">Bee</InputLabel>
          <Select required labelId="bee-label" label="Bee" style={{ backgroundColor: 'white', 'margin-bottom': '20px' }} value={beeId} onChange={(e) => { setBeeId(e.target.value); }}>
            {bees.map((bee) => {
              if (bee.id > 4) {
                return (
                  <MenuItem value={bee.id} key={bee.id}>
                    {bee.name}
                  </MenuItem>
                );
              }
              return null;
            })}
            {empty && <MenuItem value="">Cannot delete default bees</MenuItem>}
          </Select>
        </FormControl>
        <Button
          type="submit"
          variant="contained"
          color="error"
          startIcon={<RemoveCircleIcon />}
          sx={{
            fontWeight: 'bold', marginTop: '20px', width: '300px', alignSelf: 'center',
          }}
        >
          Delete Bee
        </Button>
      </form>
    </div>
  );
};
export default DeleteBeePage;
