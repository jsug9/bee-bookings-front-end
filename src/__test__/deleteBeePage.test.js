import axios from 'axios';
import { act } from 'react-dom/test-utils';
import DeleteBeePage from '../Pages/DeleteBeePage';
import renderWithProviders, { screen } from './test-utils';
import { getAllBees } from '../Redux/bees/BeesReducer';
import server from '../mswMocks/server';
import realStore from '../Redux/configureStore';

// Establish API mocking before all tests.
beforeAll(() => server.listen());
// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => server.resetHandlers());
// Clean up after the tests are finished.
afterAll(() => server.close());

it('delete bee page renders to the page', () => {
  const tree = renderWithProviders(
    <DeleteBeePage />,
  );
  expect(tree).toMatchSnapshot();
});

it('Checks that the page displays to the screen', () => {
  renderWithProviders(
    <DeleteBeePage />,
  );
  expect(screen.getByText(/Delete a Bee/)).toBeInTheDocument();
});

it('Tests the sign up functionality of the redux store', async () => {
  const store = realStore;
  const getBeesEndpoint = 'https://bee-store.herokuapp.com/api/v1/items';

  await act(async () => {
    renderWithProviders(
      <DeleteBeePage />, { store },
    );
    await store.dispatch(getAllBees());

    const deleteBee = (beeId) => {
      const data = axios.delete(`${getBeesEndpoint}/${beeId}`);
      return data;
    };

    const result = await deleteBee(1);

    expect(result.status).toEqual(200);
  });
});
