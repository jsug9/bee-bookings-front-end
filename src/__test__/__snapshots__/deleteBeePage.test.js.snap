// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete bee page renders to the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="reservation-form-container"
      >
        <form
          class="bee-form reservation-form"
          method="post"
        >
          <h1
            style="margin-bottom: 30px;"
          >
            Delete a Bee
          </h1>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              id="bee-label"
            >
              Bee
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              required=""
              style="background-color: white; margin-bottom: 20px;"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="bee-label"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                required=""
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Bee
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-12ovm6i-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="RemoveCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
                />
              </svg>
            </span>
            Delete Bee
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="reservation-form-container"
    >
      <form
        class="bee-form reservation-form"
        method="post"
      >
        <h1
          style="margin-bottom: 30px;"
        >
          Delete a Bee
        </h1>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            id="bee-label"
          >
            Bee
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            required=""
            style="background-color: white; margin-bottom: 20px;"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="bee-label"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              required=""
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Bee
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-12ovm6i-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
              />
            </svg>
          </span>
          Delete Bee
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`login page renders to the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="reservation-form-container"
      >
        <form
          class="bee-form reservation-form"
          method="post"
        >
          <h1
            style="margin-bottom: 30px;"
          >
            Delete a Bee
          </h1>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              id="bee-label"
            >
              Bee
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              required=""
              style="background-color: white; margin-bottom: 20px;"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="bee-label"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                required=""
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Bee
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-12ovm6i-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="RemoveCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
                />
              </svg>
            </span>
            Delete Bee
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="reservation-form-container"
    >
      <form
        class="bee-form reservation-form"
        method="post"
      >
        <h1
          style="margin-bottom: 30px;"
        >
          Delete a Bee
        </h1>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            id="bee-label"
          >
            Bee
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            required=""
            style="background-color: white; margin-bottom: 20px;"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="bee-label"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              required=""
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Bee
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-12ovm6i-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
              />
            </svg>
          </span>
          Delete Bee
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
